### PR TITLE
New package: py-azure-storage-blob

### PR DESCRIPTION
**Pull request by marcusboden**
depends on #28208
depends on #28207

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-azure-storage-blob/package.py
Comment (alalazo) [inline var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]: Our package audits are reporting a few inconsistencies:

Screenshot from 2022-01-03 10-52-38

Basically, this package has constraints that cannot be solved with the versions Spack knows about. Can you add proper versions to dependencies too?

**Review by alalazo — DISMISSED** [review]

File: var/spack/repos/builtin/packages/py-azure-storage-blob/package.py
Comment (marcusboden) [reply var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]: I have two PRs for these versions: #28207  and  #28208
I was planning to wait until they are merged. Or is there a better way?

Comment (alalazo) [reply var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]: @marcusboden I see that these are already in #28208 and #28207 I made the dependency explicit in the description, feel free to submit a single PR next time.

Comment (marcusboden) [reply var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]: Will do! Thanks for the feedback!

Comment (adamjstewart) [reply var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]: For future reference, we used to require that each package be updated in a separate PR, but we've relaxed this constraint for PRs that directly depend on each other. So you're welcome to continue to submit each change in separate PRs, but you no longer have to.

File: var/spack/repos/builtin/packages/py-azure-storage-blob/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]:
```suggestion
    depends_on('py-azure-core@1.10:1', type=('build', 'run'))
```

File: var/spack/repos/builtin/packages/py-azure-storage-blob/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]:
```suggestion
    depends_on('py-futures', type=('build', 'run'), when='^python@:2')
```

File: var/spack/repos/builtin/packages/py-azure-storage-blob/package.py
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-azure-storage-blob/package.py]:
```suggestion
    depends_on('py-azure-storage-nspkg@3', type=('build', 'run'), when='^python@:2')
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**
